packages/lesswrong/components/linkPreview/HoverPreviewLink.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Components, registerComponent, parseRoute, Utils } from 'meteor/vulcan:core';
+import { Link } from 'react-router-dom';
+import { hostIsOffsite, useLocation } from '../../lib/routeUtil';
+
+// From react-router-v4
+var parsePath = function parsePath(path) {
+  var pathname = path || '/';
+  var search = '';
+  var hash = '';
+  
+  var hashIndex = pathname.indexOf('#');
+  if (hashIndex !== -1) {
+    hash = pathname.substr(hashIndex);
+    pathname = pathname.substr(0, hashIndex);
+  }
+  
+  var searchIndex = pathname.indexOf('?');
+  if (searchIndex !== -1) {
+    search = pathname.substr(searchIndex);
+    pathname = pathname.substr(0, searchIndex);
+  }
+  
+  return {
+    pathname: pathname,
+    search: search === '?' ? '' : search,
+    hash: hash === '#' ? '' : hash
+  };
+};
+
+const HoverPreviewLink = ({innerHTML, href}) => {
+  // FIXME: This is currently client-side-only because 'URL' is a browser-API
+  // class. See the workaround for the same issue in PostsPage.
+  const location = useLocation();
+  
+  // Within-page relative link?
+  if (href.startsWith("#")) {
```

Comment:
```suggestion
  if (href?.startsWith("#")) {
```

Reply:
(this crashed on my local server when I had a fake or bad link in a post)

Reply:
(might want to add more defensive programming in the various dot-donation variables)